### PR TITLE
Docmosis Integration

### PR DESCRIPTION
**Pull request by rajatkgupta29**
**Before creating a pull request make sure that:**

- [ X] commit messages are meaningful and follow good commit message guidelines
- [ X] README and other documentation has been updated / added (if needed)
- [ X] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ADOP-184

### Change description ###
Addition of docmosis template integration code

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java [hunk 1]
```diff
@@ -185,6 +188,14 @@
...
     private String applicant2SotFullName;
 
+    @CCD(
+        label = "Documents generated",
+        typeOverride = Collection,
+        typeParameterOverride = "AdoptionDocument",
+        access = {DefaultAccess.class}
+    )
+    private List<ListValue<AdoptionDocument>> documentsGenerated;
+
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java:198]: change DefaultAccess.class to CollectionAccess.class

**Review by gauravtomar-gh — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/adoption/document/DocAssemblyService.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        );
+
+        log.info("Document successfully generated and stored for case Id {} with document location {}",
+            caseId,
+            docAssemblyResponse.getRenditionOutputLocation()
+        );
+
+        return new DocumentInfo(
+            docAssemblyResponse.getRenditionOutputLocation(),
+            filename + ".pdf",
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/document/DocAssemblyService.java:63]: Use constant in DocumentConstants.java for pdf & PDF and use it

**Review by pankajhmcts — CHANGES_REQUESTED** [review]
Looks good few minor change required

File: src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java (hunk 1 above)
Comment (rajatkgupta29) [reply src/main/java/uk/gov/hmcts/reform/adoption/adoptioncase/model/CaseData.java:209]: Resolved

File: src/main/java/uk/gov/hmcts/reform/adoption/document/content/DocmosisTemplateProvider.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package uk.gov.hmcts.reform.adoption.document.content;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.adoption.common.config.DocmosisTemplatesConfig;
+import uk.gov.hmcts.reform.adoption.document.model.LanguagePreference;
+
```
Comment (pankajhmcts) [inline src/main/java/uk/gov/hmcts/reform/adoption/document/content/DocmosisTemplateProvider.java:7]: Add JUnit for this class. I can see you already added Junit for most of the classes, please make sure coverage should be 90% at least 
@rajatkgupta29 @abidfs @dilipsamrahmcts @gauravtomar-gh

**Review by rajatkgupta29 — COMMENTED** [review]
Resolved

**Review by pankajhmcts — APPROVED** [review]

**Pull request merged**